path.go

```diff
@@ -370,3 +505,142 @@ func RemoveEscapeChar(word string) string {
 	}
 	return word
 }
+
+func getParamConstraintType(constraintPart string) TypeConstraint {
+	switch constraintPart {
+	case ConstraintInt:
+		return intConstraint
+	case ConstraintBool:
+		return boolConstraint
+	case ConstraintFloat:
+		return floatConstraint
+	case ConstraintAlpha:
+		return alphaConstraint
+	case ConstraintGuid:
+		return guidConstraint
+	case ConstraintMinLen:
+		return minLenConstraint
+	case ConstraintMaxLen:
+		return maxLenConstraint
+	case ConstraintExactLen:
+		return exactLenConstraint
+	case ConstraintBetweenLen:
+		return betweenLenConstraint
+	case ConstraintMin:
+		return minConstraint
+	case ConstraintMax:
+		return maxConstraint
+	case ConstraintRange:
+		return rangeConstraint
+	case ConstraintDatetime:
+		return datetimeConstraint
+	case ConstraintRegex:
+		return regexConstraint
+	default:
+		return noConstraint
+	}
+
+}
+
+func (c *Constraint) CheckConstraint(param string) bool {
+	var err error
+	var num int
+
+	// check data exists
+	needOneData := []TypeConstraint{minLenConstraint, maxLenConstraint, exactLenConstraint, minConstraint, maxConstraint, datetimeConstraint, regexConstraint}
+	needTwoData := []TypeConstraint{betweenLenConstraint, rangeConstraint}
+
+	for _, data := range needOneData {
+		if c.ID == data && len(c.Data) == 0 {
+			return false
+		}
+	}
+
+	for _, data := range needTwoData {
+		if c.ID == data && len(c.Data) < 2 {
+			return false
+		}
+	}
+
+	// check constraints
+	switch c.ID {
+	case intConstraint:
+		_, err = strconv.Atoi(param)
+	case boolConstraint:
+		_, err = strconv.ParseBool(param)
+	case floatConstraint:
+		_, err = strconv.ParseFloat(param, 32)
+	case alphaConstraint:
+		for _, r := range param {
+			if !unicode.IsLetter(r) {
+				return false
+			}
+		}
+	case guidConstraint:
+		_, err = uuid.Parse(param)
+	case minLenConstraint:
+		data, _ := strconv.Atoi(c.Data[0])
+
+		if len(param) < data {
+			return false
+		}
+	case maxLenConstraint:
+		data, _ := strconv.Atoi(c.Data[0])
+
+		if len(param) > data {
+			return false
+		}
+	case exactLenConstraint:
+		data, _ := strconv.Atoi(c.Data[0])
+
+		if len(param) != data {
+			return false
+		}
+	case betweenLenConstraint:
+		data, _ := strconv.Atoi(c.Data[0])
+		data2, _ := strconv.Atoi(c.Data[1])
+		length := len(param)
+		if length < data || length > data2 {
+			return false
+		}
+	case minConstraint:
+		data, _ := strconv.Atoi(c.Data[0])
+		num, err = strconv.Atoi(param)
+
+		if num < data {
+			return false
+		}
+	case maxConstraint:
+		data, _ := strconv.Atoi(c.Data[0])
+		num, err = strconv.Atoi(param)
+
+		if num > data {
+			return false
+		}
+	case rangeConstraint:
+		data, _ := strconv.Atoi(c.Data[0])
+		data2, _ := strconv.Atoi(c.Data[1])
+		num, err = strconv.Atoi(param)
+
+		if num < data || num > data2 {
+			return false
+		}
+	case datetimeConstraint:
+		_, err = time.Parse(c.Data[0], param)
+	case regexConstraint:
+		if c.RegexCompiler == nil {
```

Comment:
why not precompile the regex when the route is specified in the registers

if then the requests are parsed this goes on the performance